### PR TITLE
Blocks: Add codeHandler to complement pasteHandler

### PR DESCRIPTION
**Pull request by mcsf**
Alternative to #29157.

## Description
Pasting text sometimes breaks related CSS, this refactor fixes that.

## Test plan
* Create a new HTML block
* Paste `<p>This is a <em>test</em></p>` in the editor
* Confirm there were no errors and that visually everything looks correctly

## Types of changes
Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: packages/blocks/src/api/raw-handling/code-handler.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { parseWithGrammar } from '../parser';
+
+export function codeHandler( { HTML = '' } ) {
+	const freeformName = getFreeformContentHandlerName();
+	const unregisteredName = getUnregisteredTypeHandlerName();
+
+	return parseWithGrammar( HTML ).map( ( block ) => {
+		const { name } = block;
+
+		// Let TinyMCE handle it
```
Comment (ellatrix) [inline packages/blocks/src/api/raw-handling/code-handler.js:30]: Is this still necessary? Maybe we should just do it for all blocks if we only strip script and on* attributes. The freeform block could be something other than TinyMCE.

**Review by ellatrix — APPROVED** [review]
Code-wise, looks good to me.

**Comment by ellatrix** [comment]
> ## Test plan
> * Create a new HTML block
> * Paste `<p>This is a <em>test</em></p>` in the editor
> * Confirm there were no errors and that visually everything looks correctly

Should these test instructions be updated? I don't see the relevance or an error when I try this.

File: packages/blocks/src/api/raw-handling/code-handler.js (hunk 1 above)
Comment (mcsf) [reply packages/blocks/src/api/raw-handling/code-handler.js:30]: My thinking was to really minimise back-compat issues, and that may or may not include preserving even those tags. However,

> The freeform block could be something other than TinyMCE.

I did wonder about this. I considered the alternative, which was to hardcode `if ( name === 'core/freeform' )`... Worse? Better?

Comment (ellatrix) [reply packages/blocks/src/api/raw-handling/code-handler.js:30]: I'm just not sure why it should be an exception. Great if TinyMCE already does some sanitising, but it doesn't hurt to run our own?

**Pull request closed without merging**